File: abc/abc310/a/main.cpp
#include <iostream>
#include <climits>  // INT_MAXを使用するために必要
using namespace std;

int main(void)
{
    int n, p, q;
    cin >> n >> p >> q;

    int d_min = INT_MAX, d;
    for(int i = 1; i <= n; i++){
        cin >> d;
        d_min = min(d, d_min);
    }
    cout << min(p, q+d_min) << endl;
    
    return 0;
}

File: abc/abc310/a/main.go
package main

import (
	"fmt"
	"math"
)

func main() {
	var n, p, q int
	fmt.Scan(&n, &p, &q)

	dMin := math.MaxInt32
	var d int
	for i := 1; i <= n; i++ {
		fmt.Scan(&d)
		dMin = int(math.Min(float64(d), float64(dMin)))
	}

	fmt.Println(min(p, q+dMin))
}

func min(a, b int) int {
	if a < b {
		return a
	}
	return b
}
